### DISCUSSION TITLE
Navigation

### DISCUSSION THREAD

**Discussion by spykee2005** [BimmerGestalt/AAIdrive#547] [Q&A]
Hi all,

Thnak you for making this thing happen. I was desperate for not be able to use my android phone in my car.

I have a 2019 BMW X3 G01 with Idrive6. How is it possible to use Google Maps navigation instead of the inbuild nav system?

Thx a lot and keep doing!!! 

A thankful user!!

**Comment by hufman** [comment]
Thank you for the kind words!
Unfortunately, AAIdrive doesn't provide a replacement for the car's nav system: Instead, it can provide a better-than-nothing map for cars that don't include any map at all. There is a [beta build](https://bimmergestalt.s3.amazonaws.com/aaidrive/builds/androidautoidrive-latest-main-mapbox-sentry-release.apk) with a custom map implementation, based on Mapbox maps and routing data, but it only draws a blue line on the map instead of trying to replicate a full navigation experience.
There may be a future feature to enable a next-turn display #539, but I'm not actually sure how to locate the user within the route instructions in order to show the next instruction to display. Contributions are welcome: This part of code is relatively separated from the car protocol, and is merely parsing the Mapbox API and drawing to an Android layout.

There is an option for users to compile a version that uses Google Maps for the map and routing information, but it does not provide any extra navigation functionality. However, the Google Maps license terms specifically disallow trying to replicate the Google Maps navigation experience using their data. The license also disallows showing Google Maps in the car, which is why this version is not available in the public build.

The closest you'll be able to get is to use the Screen Mirroring addon (or code your car to enable Miracast) to show the official Google Maps on the car screen, but your phone screen has to stay on and there isn't any input support from the car. Unfortunately, there isn't any other way to use Google Maps navigation in IDrive 6 from an Android phone.

**Reply by spykee2005** [reply]
Thanks for the reply. Is there a possibility to increase the refresh rate to make the navigation much smoother?

**Reply by hufman** [reply]
The Bluetooth connection is very low bandwidth, and I tried to compromise on the side of update frequency by setting such a low quality for the JPG compression. The only other option is to run the apps connection over USB.
